src/dex-router.ts

```diff
@@ -111,25 +114,17 @@ export class DexRouter {
       `1inch quote: ${amount.toString()} ${tokenIn} -> ${quoteResult.dstAmount} ${tokenOut}`
     );
 
-    const params: {
-      fromTokenAddress: string;
-      toTokenAddress: string;
-      amount: string;
-      fromAddress: string;
-      slippage: number;
-      connectorTokens?: string;
-    } = {
+    const params = {
       fromTokenAddress: tokenIn,
       toTokenAddress: tokenOut,
       amount: amount.toString(),
       fromAddress,
       slippage,
+      ...(this.connectorTokens.length > 0
+        ? { connectorTokens: this.connectorTokens }
+        : {}),
     };
 
-    if (this.connectorTokens.length > 0) {
-      params['connectorTokens'] = this.connectorTokens;
-    }
```

Comment:
I kind of liked this way of doing it better.  In my branch I've added another parameter here, and seems cleaner.